### ISSUE TITLE
ReScript...

### ISSUE THREAD

**Issue by MoOx** [rescript-react-native/rescript-react-native#731]
Hi everyone,

Here is a few question for you, do not hesitate to comment or use reactions.

- 👍 / 👎  Now that we require ~~BuckleScript~~ ReScript 8.2 (bs-platform, soon to be renamed to match rescript rebranding), should we convert the codebase to new ReScript syntax? (Is that possible for all bindings btw ? I didn't checked what *isn't* possible with the new syntax) - I think we can even script that part !
- 😄 / 😕  Do we want to rename this organisation and all repo here and on npm to `rescript-*` ?
- 🚀 / 👀 Should we do this now 🚀 (like NOW (republish latest version of current packages with new names) or for 0.64 release (what about other packages?) or should we wait for bs-platform being renamed 👀 ? 
- ❤️ / 🎉 Also for the website I want to do some dogfooding (for fun) but maybe we should go for a docusaurus... ❤️(keep homemade)  / 🎉 (go docusaurus)?

**Comment by cknitt** [comment]
* Conversion to ReScript syntax: Yes, I think we should definitely do that, the only question is when... The ReScript parser/printer are probably good enough already, but I would wait until the editor integration is ready, too.
* Rename: Yes, IMHO we should rename everything to `rescript-*` (although that's going to be a lot of work). I think we can wait until bs-platform is renamed though.

**Comment by gasacchi** [comment]
i think is good idea to use new .res syntax, since they already got editor support :100:

**Comment by fakenickels** [comment]
Another breaking change with no backwards compatibility https://github.com/rescript-lang/rescript-react

**Comment by MoOx** [comment]
I guess we are going to do this changes for 0.64.

**Comment by MoOx** [comment]
The approach I will take:

- upgrade to 0.64
- publish as reason-react-native
- upgrade to rescript the same version (might introduce breaking changes for some polyvariant)
- publish as rescript-react-native

**Comment by Freddy03h** [comment]
Does having both `reason-react-native@0.64` & `rescript-react-native@0.64` mean maintaining two repositories or branches ?

**Comment by MoOx** [comment]
Nope, we will just have 2 packages on npm but a single repo named rescript-*

**Comment by MoOx** [comment]
(reason-react-native@0.64 will published but deprecated directly on npm with a warning to the new name)

**Comment by MoOx** [comment]
0.64 is out... Starting to flip the table!

**Comment by MoOx** [comment]
You can try `rescript-react-native@0.64.2` and see how it goes. I will update all packages of this repo in the coming days (most work is done locally, I have to review, push & publish)

**Comment by MoOx** [comment]
I will deprecated reason-* later on, when things looks good with rescript-* packages.

**Comment by MoOx** [comment]
Following what I will call "community management issues" I am going to pause this migration for a bit.

**Comment by MoOx** [comment]
Website WIP https://rescript-react-native.github.io
Dogfooding was fun but generally:
- We update rrn bindings
- we should update website at the same time
- package sometimes needs update (all the one that depends on rrn or that "forwards" View or Text props... - so a lot) this take some time - even if partly automated
- so website is not updated because website use rrn & some bindings (dogfooding at max)

I decided to stop this because it's consuming me too much unecessary times.
Especially that now I have for my concern multiples projects to test the bindings on.

So now, a simple Docusaurus, with all markdown files there https://github.com/rescript-react-native/rescript-react-native.github.io/tree/src/docs
Exactly like RN website...

If anyone want to contribute on the website this will be simpler.

Next: updating all package to the org rebranding...

**Comment by MoOx** [comment]
Open for more serious testing!

**I migrated all packages of this org this morning.**

(template has been updated to 0.64 as well - I have a minor glitch to fix but it should be out on npm soon).

2 things before I close this issue:
- add algolia to the website
- write a blog post to communicate about this work to let know people that React Native can now safely be used with ReScript

One additional thing will be to replace bs-platform in all package with ReScript when it's CLI is stable.
